jOOQ/src/main/java/org/jooq/impl/BatchSingle.java

```diff
@@ -200,7 +200,7 @@ private final void checkBindValues() {
         // [#4062] Make sure we collect also repeated named parameters
         ParamCollector collector = new ParamCollector(configuration, false);
         collector.visit(query);
-        List<Param<?>> params = new ArrayList<Param<?>>();
+        List<Param<?>> params = new ArrayList<Param<?>>(collector.resultList.size());
```

Comment:
Another candidate for an array (I'll just fix these. Further comments will be implied)